automation_tools/__init__.py

```diff
@@ -2127,6 +2130,151 @@ def get_hostname_from_ip(ip, timeout=3):
+    already.
+
+    :param instance_name: A string. RHEVM Instance name to create.
+    :param template_name: A string. RHEVM image name from which instance
+        to be created.
+    :param int timeout: The polling timeout in minutes to create rhevm
+    instance.
+
+    ssh_key should be added to openstack project before running automation.
+    Else the automation will fail.
+
+    The following environment variables affect this command:
+
+    USERNAME
+        The username of a rhevm project to login.
+    PASSWORD
+        The password of a rhevm project to login.
+    API_URL
+        An url to API of rhevm project.
+
+    """
+    obj = get_rhevm_client()
+    temp = obj.templates.get(name=template_name)
+    dc = obj.datacenters.get(name='Default')
+    quota = dc.quotas.get(name='admin')
+    print('Turning on instance {0} from template {1}. Please wait'
+          'till it get up ...'.format(instance_name, template_name))
+    obj.vms.add(
+        params.VM(
+            name=instance_name,
+            cluster=obj.clusters.get('Default'),
+            template=temp, quota=quota))
+    timeup = time.time() + int(timeout) * 60
+    while True:
+        if time.time() > timeup:
+            print('The timeout for creating RHEVM instance has reached!')
+            sys.exit(1)
+        vm_status = obj.vms.get(name=instance_name).get_status().get_state()
+        print(vm_status)
+        if vm_status == 'down':
+            obj.vms.get(name=instance_name).start()
+            while True:
+                if time.time() > timeup:
+                    print('The timeout for creating RHEVM instance is reached')
+                    sys.exit(1)
+                vm_status = obj.vms.get(
+                    name=instance_name).get_status().get_state()
+                print(vm_status)
+                if vm_status == 'up':
+                    break
+                time.sleep(5)
+            break
+        time.sleep(3)
+    print('Instance {0} is now UP !'.format(instance_name))
+    obj.disconnect()
+
+
+def delete_rhevm_instance(instance_name, timeout=5):
+    """Deletes RHEVM Instance.
+
+    :param instance_name: A string. RHEVM instance name to delete.
+    :param int timeout: The polling timeout in minutes to delete rhevm
+    instance.
+
+    The following environment variables affect this command:
+
+    USERNAME
+        The username of a rhevm project to login.
+    PASSWORD
+        The password of a rhevm project to login.
+    API_URL
+        An url to API of rhevm project.
+    """
+    obj = get_rhevm_client()
+    vm = obj.vms.list(query='name={0}'.format(instance_name))
+    if not vm:
+        print('The instance {0} is not found '
+              'in RHEVM to delete!'.format(instance_name))
+    else:
+        print('Deleting instance {0} from RHEVM.....'.format(instance_name))
+        timeup = time.time() + int(timeout) * 60
+        if obj.vms.get(name=instance_name).get_status().get_state() == 'up':
```

Comment:
We should try to make this code simpler. I'm surprised that the client don't handles the deletion and a pooling is needed.

I will try to think on a better way to handle this, but at first I think the waiting for something can be abstracted into a function for waiting for something to happen.